src/main/java/bio/terra/cli/WorkspaceUser.java

```diff
@@ -0,0 +1,117 @@
+package bio.terra.cli;
+
+import bio.terra.cli.exception.UserActionableException;
+import bio.terra.cli.service.WorkspaceManagerService;
+import bio.terra.workspace.model.IamRole;
+import bio.terra.workspace.model.RoleBindingList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Internal representation of a workspace user (i.e. someone a workspace is shared with). This is
+ * different from a regular {@link bio.terra.cli.User} because they are never logged in. This is
+ * just a reference to another Terra user who has some level of workspace access. This class is not
+ * part of the current context or state.
+ */
+public class WorkspaceUser {
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceUser.class);
+
+  private String email;
+  private List<IamRole> roles;
+
+  private WorkspaceUser(String email, List<IamRole> roles) {
+    this.email = email;
+    this.roles = roles;
+  }
+
+  /**
+   * Add a user to the current workspace. Possible roles are defined by the WSM client library.
+   *
+   * @param email email of the user to add
+   * @param role role to assign the user
+   * @throws UserActionableException if there is no current workspace
+   */
+  public static WorkspaceUser add(String email, IamRole role) {
+    Workspace currentWorkspace = Context.requireWorkspace();
+
+    // call WSM to add a user + role to the current workspace
+    new WorkspaceManagerService().grantIamRole(currentWorkspace.getId(), email, role);
+    logger.info("Added user to workspace: user={}, role={}", email, role);
+
+    // return a WorkspaceUser that includes all roles this email has
```

Comment:
nit: comment makes me worried that we sometimes return WorkspaceUsers that *don't* include all roles.

Reply:
re-worded comment. the methods in this class always return a `WorkspaceUser` object that includes the email + all roles that email has.